### PR TITLE
Linux edge fixes

### PR DESCRIPTION
**Pull request by luke-hill**
* Allow Linux Edge to be automatically detected and interrogated for browser version
* Allow msedgedriver to be automatically detected and downloaded on Linux
* Prevent NME caused by selenium versions not being fully updated

### CODE REVIEW

File: lib/webdrivers/edge_finder.rb [hunk 1]
```diff
@@ -24,6 +24,8 @@ def location
       private
 
       def user_defined_location
+        return unless defined? Selenium::WebDriver::Edge.path
```
Comment (titusfortner) [inline lib/webdrivers/edge_finder.rb:27]: Why do you need this line? If it isn't defined it probably isn't going to do the right thing.

Comment (luke-hill) [reply lib/webdrivers/edge_finder.rb:27]: This permits people using this gem and slightly older versions of selenium. This way you don't get crashes, it just fails gracefully and then attempts to find the location using the System platform location methods.

Comment (titusfortner) [reply lib/webdrivers/edge_finder.rb:27]: The versions of Selenium that don't have this method don't support Chromium Edge, so it shouldn't matter. Do you have a specific scenario where an error was raised, but getting a nil would have done the right thing?

Comment (luke-hill) [reply lib/webdrivers/edge_finder.rb:27]: Still using 4alpha6. Think I've mentioned. This prevents a NME

Comment (titusfortner) [reply lib/webdrivers/edge_finder.rb:27]: Hah, I thought `#path` had been in there longer. It was added for alpha 7. You should be able to add this and have it work in alpha 6:
```
Selenium::WebDriver::Edge = Selenium::WebDriver::EdgeChrome
```

Comment (luke-hill) [reply lib/webdrivers/edge_finder.rb:27]: Yep, but is there any harm in adding this as a preventative measure, given it doesn't change what the function does.

**Comment by titusfortner** [comment]
I uncommented @kapoorlakshya's code for 4.7
5.0 is going to require Selenium 4+ so the conditional won't be needed

**Comment by luke-hill** [comment]
Awesome stuff thanks!

**Pull request closed without merging**